flytekit/core/type_engine.py

```diff
@@ -290,17 +289,19 @@ class TypeEngine(typing.Generic[T]):
     _DATACLASS_TRANSFORMER: TypeTransformer = DataclassTransformer()
 
     @classmethod
-    def register(cls, transformer: TypeTransformer):
+    def register(cls, transformer: TypeTransformer, additional_types: typing.Optional[typing.List[Type]] = None):
         """
         This should be used for all types that respond with the right type annotation when you use type(...) function
         """
-        if transformer.python_type in cls._REGISTRY:
-            existing = cls._REGISTRY[transformer.python_type]
-            raise ValueError(
-                f"Transformer {existing.name} for type {transformer.python_type} is already registered."
-                f" Cannot override with {transformer.name}"
-            )
-        cls._REGISTRY[transformer.python_type] = transformer
+        types = [transformer.python_type, *(additional_types or [])]
```

Comment:
not needed,,just register multiple times

Reply:
actually i don't know how to do this cleanly.   Because the transformer specifies the python type when it inherits from the base transformer: `class FlyteFilePathTransformer(TypeTransformer[FlyteFile])`, I'd have to create a base class out of the transformer, and then create two child classes out of it I think.  I think this is actually cleaner.